src/MagnumPlugins/AssimpImporter/Test/AssimpImporterTest.cpp

```diff
@@ -1942,38 +2141,27 @@ void AssimpImporterTest::scene() {
     CORRADE_COMPARE(importer->object3DForName("Ghost"), -1);
 }
 
-void AssimpImporterTest::sceneCollapsedNode() {
+void AssimpImporterTest::sceneNoCollapsedNode() {
     Containers::Pointer<AbstractImporter> importer = _manager.instantiate("AssimpImporter");
 
-    /* This collapses all nodes into one. Neither OptimizeGraph nor
-       OptimizeMeshes does that, but this one does it. Um. */
+    /* Without IMPORT_NO_SKELETON_MESHES, this collapses all nodes into one.
+       AssimpImporter unconditionally sets this so this shouldn't happen. Neither
+       OptimizeGraph nor OptimizeMeshes does that, but this one does it. Um. */
     importer->configuration().group("postprocess")->setValue("PreTransformVertices", true);
 
     CORRADE_VERIFY(importer->openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR, "scene.dae")));
 
     CORRADE_COMPARE(importer->defaultScene(), 0);
     CORRADE_COMPARE(importer->sceneCount(), 1);
-    CORRADE_COMPARE(importer->object3DCount(), 1); /* Just the root node */
+    CORRADE_COMPARE(importer->object3DCount(), 2);
 
     Containers::Optional<SceneData> scene = importer->scene(0);
     CORRADE_VERIFY(scene);
     CORRADE_COMPARE(scene->children3D(), {0});
 
-    /* Assimp makes some bogus mesh for this one */
-    Containers::Pointer<ObjectData3D> collapsedNode = importer->object3D(0);
-    CORRADE_COMPARE(collapsedNode->children(), {});
-    CORRADE_COMPARE(collapsedNode->instanceType(), ObjectInstanceType3D::Mesh);
-    CORRADE_COMPARE(collapsedNode->transformation(), Matrix4{});
-
-    /* Name of the scene is used for the root object */
-    {
-        const UnsignedInt version = aiGetVersionMajor()*100 + aiGetVersionMinor();
-        /** @todo Possibly works with other versions (definitely not 3.0) */
-        CORRADE_EXPECT_FAIL_IF(version <= 302,
-            "Assimp 3.2 and below doesn't use name of the root node for collapsed nodes.");
-        CORRADE_COMPARE(importer->object3DForName("Scene"), 0);
-        CORRADE_COMPARE(importer->object3DName(0), "Scene");
-    }
+    Containers::Pointer<ObjectData3D> parent = importer->object3D(0);
+    CORRADE_COMPARE(parent->children(), {1});
+    CORRADE_COMPARE(parent->instanceType(), ObjectInstanceType3D::Empty);
```

Comment:
Ummm ... I don't understand why is there such a weird interaction with `IMPORT_NO_SKELETON_MESHES`?

IIRC the original intent of this test was to verify that collapsing the whole file into a single node works, i.e. testing corner cases of [this block of code](https://github.com/mosra/magnum-plugins/blob/faa16d460d21dc8eb3682aec4335954db5dca758/src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp#L409-L440) (and the COLLADA/skeleton-related comment there is probably outdated now?). 

I *think* in this case the expected behavior was that the importer skips the useless extra added root node and returns only its child, but I have no idea why it's no longer being done -- it should be.

The "only a single node" case in the code happens for PLY files for example (or some of the glTF files), but seems like the test doesn't explicitly verify that. I'll add a test case to check that code path.

Reply:
The comment in the test suggested it was Assimp doing the collapsing internally so I simply adapted the test. But if it tests some importer code then it might need some more scrutiny. I'm not entirely sure what's happening either, but I can do some more debugging.

Reply:
Mildly related question: what are *bone nodes*?

https://github.com/mosra/magnum-plugins/blob/faa16d460d21dc8eb3682aec4335954db5dca758/src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp#L571

Reply:
Outdated comment -- nodes associated with a skin. Basically what's provided by `SkinData` now, I suppose this TODO expected the reference would go the other way around. Remove :)

> Assimp doing the collapsing internally so I simply adapted the test.

IIRC Assimp does the collapsing internally if you enable PreTransformVertices, *but* still adds a silly empty root node with an identity transformation that the importer plugin then attempts to remove.